FOCUS #430: EffectiveCost spec change - addressing purchases paid to cover future eligible charges (#431)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/columns/effectivecost.md

# Effective Cost

> Effective Cost represents a cost inclusive of the impacts of all reduced rates and discounts, augmented with the [*amortization*](#glossary:amortization) of relevant purchases (one-time or recurring) paid to cover future eligible [*charges*](#glossary:charge), with the exception that, in the case of a purchase charge paid to cover future eligible charges, it amounts to 0. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. Effective Cost does not mix or "blend" costs across multiple charges of the same service. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.

Reviewer: I would consider this more of a clarification than an "exception"

"augmented with the amortization of relevant purchases" to me means that in the case of a purchase of a pre-paid expense (e.g. an upfront Reserved Instance) EffectiveCost would include the negative amortization (which is technically a capitalization of the cost) resulting in the EffectiveCost netting to $0

This is not a blocking comment, just sharing my perspective.

Reviewer 2: @kk09v I just wrote the same thing above and didn't see your comment first haha

Reviewer 3: I agree this might be clearer as a clarification rather than an exception.

I hate to expand the scope of the change, but would it be possible to simplify a little here? I'm getting hung up on the first half of the sentence. While accurate, it's a bit hard for me to grok. I have to read it multiple times and strain a bit. Amortization purists might say this isn't completely accurate, but attempting just in case...

Here's what I want to say in the simplest, most direct way:
```suggestion
Effective Cost represents the [*amortized*](#glossary:amortization) cost of the [*charge*](#glossary:charge) after applying all reduced rates, discounts, and the applicable portion of relevant, prepaid purchases (one-time or recurring) that covered this charge. Since amortization breaks down and spreads the original purchase, Effective Cost is always 0 for prepaid purchase charges that were not covered by another purchase. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. Effective Cost does not mix or "blend" costs across multiple charges of the same service. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.
```

Here's a possible middle ground:
```suggestion
Effective Cost represents the cost of the [*charge*](#glossary:charge) after applying all reduced rates, discounts, and any [*amortized*](#glossary:amortization) charges from relevant purchases (one-time or recurring) paid to cover future eligible changes. Since amortization breaks down and spreads the original purchase, Effective Cost is always 0 for prepaid purchase charges that were not covered by another purchase. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. Effective Cost does not mix or "blend" costs across multiple charges of the same service. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.
```

Author: the first suggestion out of two included in the current version of the definition

> Effective Cost represents a cost inclusive of the impacts of all reduced rates and discounts, augmented with the [*amortization*](#glossary:amortization) of relevant purchases (one-time or recurring) paid to cover future eligible [*charges*](#glossary:charge), with the exception that, in the case of a purchase charge paid to cover future eligible charges, it amounts to 0. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. Effective Cost does not mix or "blend" costs across multiple charges of the same service. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.

Reviewer 4: ```suggestion
Effective Cost represents the [*amortized*](#glossary:amortization) cost of the [*charge*](#glossary:charge) after applying all reduced rates, discounts, and the applicable portion of relevant, prepaid purchases (one-time or recurring) that covered this charge. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. When amortization is used to spread the Billed Cost of a prepaid purchase, to future eligible charges, the Effective Cost of the purchase is set to 0. Effective Cost does not mix or "blend" costs across multiple charges of the same service. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.
```

> Effective Cost represents the [*amortized*](#glossary:amortization) cost of the [*charge*](#glossary:charge) after applying all reduced rates, discounts, and the applicable portion of relevant, prepaid purchases (one-time or recurring) that covered this charge. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. When amortization is used to spread the Billed Cost of a prepaid purchase (or related charges e.g. Tax), to future eligible charges, the Effective Cost of the original prepaid charges are set to 0. Effective Cost does not mix or "blend" costs across multiple charges of the same service. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.

Author: Are Tax charges being amortized or calculated based on related, already amortized charges?

```suggestion
Effective Cost represents the [*amortized*](#glossary:amortization) cost of the [*charge*](#glossary:charge) after applying all reduced rates, discounts, and the applicable portion of relevant, prepaid purchases (one-time or recurring) that covered this charge. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. Since amortization breaks down and spreads the cost of a prepaid purchase, to subsequent eligible charges, the Effective Cost of the original prepaid charge is set to 0.. Effective Cost does not mix or "blend" costs across multiple charges of the same service. This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.
```

Author: When EffectiveCost is set to 0, does it indicate that the charge is fully amortized? Any leftover portion would be provided in the Unused charge record?

This column resolves two challenges that are faced by practitioners:

1. Practitioners need to *amortize* relevant purchases, such as upfront fees, throughout the *commitment* and distribute them to the appropriate reporting groups (e.g. [*tags*](#glossary:tag), [*resources*](#glossary:resource)).
2. Many [*commitment-based discount*](#glossary:commitment-based-discount) constructs include a recurring expense for the *commitment* for every [*billing period*](#glossary:billing-period) and must distribute this cost to the *resources* using the *commitment*. This forces reconciliation between the initial *commitment* [*row*](#glossary:row) per period and the actual usage *rows*.

> The EffectiveCost column MUST be present in the billing data and MUST NOT be null. This column MUST be of type Decimal, MUST conform to [Numeric Format](#numericformat) requirements, and be denominated in the BillingCurrency. The EffectiveCost MUST be 0 when a charge is associated with a purchase (one-time or recurring) paid to cover future eligible charges. The aggregated EffectiveCost for a billing period may not match the charge received on the invoice for the same *billing period*.

Reviewer 3: Should we bind this to ChargeCategory? 
```suggestion
The EffectiveCost column MUST be present in the billing data and MUST NOT be null. This column MUST be of type Decimal, MUST conform to [Numeric Format](#numericformat) requirements, and be denominated in the BillingCurrency. EffectiveCost MUST be 0 when ChargeCategory is "Purchase" and the purchase is intended to cover future eligible charges. The aggregated EffectiveCost for a billing period may not match the charge received on the invoice for the same *billing period*.
```

Side note: I feel like "cover future eligible charges" is a bit abstract. We might want to revisit that post-1.0.

## Description

> Cost inclusive of the impacts of all reduced rates and discounts, augmented with the *amortization* of relevant purchases (one-time or recurring) paid to cover future eligible *charges*, with the exception that, in the case of a purchase charge paid to cover future eligible charges, it amounts to 0.

Reviewer 5: @ijurica are you trying to say effective cost must be the amortization of purchases to cover future charges but must not include the upfront cost of the purchase charge?

Reviewer 2: I like the suggestion, but I see this as a clarification statement rather than an "exception".

Maybe just make it a new sentence like:
"Cost inclusive of the impacts of all reduced rates and discounts, augmented with the *amortization* of relevant purchases (one-time or recurring) paid to cover future eligible *charges*. In the case of a purchase charge paid to cover future eligible charges, EffectiveCost amounts to 0."
